sympy/core/expr.py

```diff
@@ -1940,6 +1940,21 @@ def extract_multiplicatively(self, c):
            x/6
 
         """
+        if (type(c) == int and c < 0):
+            return Expr.extract_multiplicatively(self, c*S.ImaginaryUnit*-S.ImaginaryUnit)
```

Comment:
Well I didn't knew `Integer(c)` function and its `c*I*-I` which will equal `c`
Ok I will replace that line with `Integer(c)`